### PR TITLE
ORSP ID Passthroughs

### PR DESCRIPTION
**Pull request by rushtong**
Addresses https://broadinstitute.atlassian.net/browse/GAWB-1759

* Passthrough for retrieving a consent from an ORSP ID.
* New models for valid consent and error cases.
* This is a replacement for a closed PR: https://github.com/broadinstitute/firecloud-orchestration/pull/399

Since this touches authentication and authorization, we have a separate story to deal with FISMA changes. See https://broadinstitute.atlassian.net/browse/GAWB-1718 for addressing that aspect of this work.

Requires config changes.
* Local: https://github.com/broadinstitute/firecloud-develop/pull/338
* Dev: https://github.com/broadinstitute/firecloud-develop/pull/339

## Testing Notes:
* Update configs using https://github.com/broadinstitute/firecloud-develop/pull/338
* Spin up local orchestration
* Sign into swagger as any FC or DUOS user
* Signing into swagger with a valid OAuth, but non-FC user, will give a 401 error on any query
* Query on some ORSP IDs. Good ones include:
  * OD-310 (should be not found)
  * OD-320 (should be valid)
  * OD-330 (should be not approved) 
  * Any random string

---

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Product Owner** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge to develop
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/test/scala/org/broadinstitute/dsde/firecloud/service/LibraryApiServiceSpec.scala [hunk 1]
```diff
@@ -68,18 +69,28 @@ class LibraryApiServiceSpec extends BaseServiceSpec with LibraryApiService with
...
+    val consentError = ConsentError(message = "Unapproved", code = BadRequest.intValue)
+    val consentNotFound = ConsentError(message = "Not Found", code = NotFound.intValue)
 
-    workspaceServer = startClientAndServer(workspaceServerPort)
-    workspaceServer
-      .when(request.withMethod("GET").withPath(isCuratorPath))
-      .respond(
-        org.mockserver.model.HttpResponse.response()
-          .withHeaders(MockUtils.header).withStatusCode(OK.intValue)
-      )
```
Comment (rushtong) [inline src/test/scala/org/broadinstitute/dsde/firecloud/service/LibraryApiServiceSpec.scala:78]: This workspace server was not in use. Additionally, it was mocking an orchestration path, not a rawls path.

**Review by davidangb — APPROVED** [review]
:+1: looks good works good. Only issue I see is that with the related config changes in fc-dev, we're likely to see developers having problems running locally until they update configs; might need to shoot out an email or something.

**Review by skwalker — APPROVED** [review]
this PR is 👍, but I do think consent permissions should be fixed (on dev consent I can get a consent even if I'm not a FC or DUOS user.. )

**Comment by rushtong** [comment]
@skwalker - I'm testing the non-FC user now in dev and I do get a 401. Can you confirm that the user you are authenticating with is not in FC?

```
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>401 Unauthorized</title>
</head><body>
<h1>Unauthorized</h1>
<p>This server could not verify that you
are authorized to access the document
requested.  Either you supplied the wrong
credentials (e.g., bad password), or your
browser doesn't understand how to supply
the credentials required.</p>
<hr>
<address>Apache Server at firecloud-orchestration.dsde-dev.broadinstitute.org Port 443</address>
</body></html>
```

**Comment by skwalker** [comment]
yeah it's fine on the orch side, but on dev consent it lets anyone get the consents (I used my personal gmail which is definitely not a duos user)

**Comment by rushtong** [comment]
Ahh, that's a great point. I have a story to deal with that:
https://broadinstitute.atlassian.net/browse/GAWB-1779

**Pull request merged**